test/e2e/framework/backup_util.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2018 Oracle and/or its affiliates. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	ocicore "github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/block"
+)
+
+// CreatePVCAndBackupOrFail calls CreateAndAwaitPVCOrFail creates a new PVC based on the
+// jig's defaults, waits for it to become ready. The volume is then backed up.
+func (j *PVCTestJig) CreatePVCAndBackupOrFail(storageClient ocicore.BlockstorageClient, namespace string, volumeSize string, scName string, tweak func(pvc *v1.PersistentVolumeClaim)) (*v1.PersistentVolumeClaim, string) {
+	pvc := j.CreateAndAwaitPVCOrFail(namespace, volumeSize, scName, tweak)
+	backupVolumeID, err := j.CreateBackupVolume(storageClient, pvc)
+	if err != nil {
+		Failf("Failed to created backup for pvc %q: %v", pvc.Name, err)
+	}
+	return pvc, backupVolumeID
+}
+
+// CreateBackupVolume creates a volume backup on OCI from an exsiting volume and returns the backup volume id
+func (j *PVCTestJig) CreateBackupVolume(storageClient ocicore.BlockstorageClient, pvc *v1.PersistentVolumeClaim) (string, error) {
+	By("Creating backup of the volume")
+	pvc, err := j.KubeClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
As this isn't an `OrFail` method it should return the error rather than assert that the error has not occurred.